src/main/java/org/folio/bulkops/service/MetadataProviderService.java

```diff
@@ -1,30 +1,71 @@
+import org.folio.bulkops.domain.dto.DataImportStatus;
 import org.springframework.stereotype.Service;
 
+import java.util.List;
+import java.util.Set;
 import java.util.UUID;
 
 @Service
 @RequiredArgsConstructor
 public class MetadataProviderService {
-  private static final String COMPOSITE_PARENT = "COMPOSITE_PARENT";
+  private final Set<DataImportStatus> completedStatuses = Set.of(COMMITTED, CANCELLED, ERROR);
   private final MetadataProviderClient metadataProviderClient;
+  private final DataImportClient dataImportClient;
 
-  public DataImportJobExecution getDataImportJobExecutionByJobProfileId(UUID dataImportJobProfileId) {
-    var executions = metadataProviderClient.getJobExecutionsByJobProfileIdAndSubordinationType(dataImportJobProfileId, COMPOSITE_PARENT)
-        .getJobExecutions().stream()
-        .filter(jobExecution -> nonNull(jobExecution.getJobProfileInfo()))
-        .filter(jobExecution -> jobExecution.getJobProfileInfo().getId().equals(dataImportJobProfileId))
-        .toList();
+  public List<DataImportJobExecution> getJobExecutions(UUID jobProfileId) {
+    var splitStatus = dataImportClient.getSplitStatus();
+    var subordinationType = TRUE.equals(splitStatus.getSplitStatus()) ?
+      DataImportJobExecution.SubordinationTypeEnum.COMPOSITE_CHILD :
+      DataImportJobExecution.SubordinationTypeEnum.PARENT_SINGLE;
+    return metadataProviderClient.getJobExecutionsByJobProfileId(jobProfileId, Integer.MAX_VALUE).getJobExecutions().stream()
+      .filter(jobExecution -> nonNull(jobExecution.getJobProfileInfo()))
+      .filter(jobExecution -> jobExecution.getJobProfileInfo().getId().equals(jobProfileId))
+      .filter(jobExecution -> subordinationType.equals(jobExecution.getSubordinationType()))
+      .toList();
+  }
+
+  public boolean isDataImportJobCompleted(List<DataImportJobExecution> jobExecutions) {
+    return !jobExecutions.isEmpty() && jobExecutions.size() == jobExecutions.get(0).getTotalJobParts() && jobExecutions.stream()
+      .allMatch(jobExecution -> completedStatuses.contains(jobExecution.getStatus()));
+  }
+
+  public DataImportProgress calculateProgress(List<DataImportJobExecution> jobExecutions) {
+    var progress = new DataImportProgress().current(0).total(0);
+    jobExecutions.stream()
+      .map(DataImportJobExecution::getProgress)
+      .forEach(dataImportProgress -> {
+        progress.setCurrent(progress.getCurrent() + dataImportProgress.getCurrent());
+        progress.setTotal(progress.getTotal() + dataImportProgress.getTotal());
+      });
+    return progress;
+  }
 
-    return executions.isEmpty() ?
-      new DataImportJobExecution().progress(new DataImportProgress().current(0).total(0)) :
-      executions.get(0);
+  public List<String> getUpdatedInstanceIds(List<DataImportJobExecution> jobExecutions) {
+    return jobExecutions.stream()
+      .map(DataImportJobExecution::getId)
+      .map(uuid -> metadataProviderClient.getJobLogEntries(uuid.toString(), Integer.MAX_VALUE))
```

Comment:
hoping for case when splitting is disabled, it's not possible to import many records, otherwise it will be big issue to retrieve 20-30-40k